Automattic/jetpack
# Update Sync Queue so that serialize is wrapped to catch errors

Serialize can throw exceptions when called on closures. This update catches serialization errors so that the action/data fails silently allowing continued processing.

Fixes #7252 

#### Changes proposed in this Pull Request:
* Handle Fatal exception caused by anonymous functions on callables.

#### Does this pull request change what data or activity we track or use?
* NO

#### Testing instructions:

* On Master branch
* Add a hook to one of Sync's callable where the function internally uses an anonymous function. functions.https://github.com/Automattic/jetpack/blob/e13487d0dbbe4c743739b6f1a3e37c143a8c2614/projects/packages/sync/src/class-defaults.php#L278
* do_action( 'jetpack_sync_unlock_sync_callable' );
* perform an action (edit post)
* Fatal Exception :  Uncaught Exception: Serialization of 'Closure' is not allowed should be triggered
* Checkout fix/sync-serialize-closures branch
* do_action( 'jetpack_sync_unlock_sync_callable' );
* perform an action (edit post)
* Verify no fatal exceptions

#### Proposed changelog entry for your changes:
* Bug Fix : Catch serialization issues in Jetpack Sync.
